sha3/sha3.go

```diff
@@ -0,0 +1,27 @@
+package sha3
+
+// Drop-in replacement for golang.org/x/crypto/sha3
+// Assumes openssl.Init() was called
+
+import (
+	"crypto"
+	"github.com/golang-fips/openssl"
+)
+
+var (
+	New224 = openssl.NewSHA3_224
+	New256 = openssl.NewSHA3_256
+	New384 = openssl.NewSHA3_384
+	New512 = openssl.NewSHA3_512
+	Sum224 = openssl.SHA3_224
+	Sum256 = openssl.SHA3_256
+	Sum384 = openssl.SHA3_384
+	Sum512 = openssl.SHA3_512
+)
```

Comment:
The Go compiler can't apply many important optimizations (e.g., inlining and escape analysis) when calling a function assigned to a `var`, as it's value can change at runtime.

Better define a wrapper functions, which won't have a performance impact as it will be inlined away.

```go
func New224() hash.Hash {
    return openssl.NewSHA3_224()
}
```